### PR TITLE
feat: add serialize formating

### PR DESCRIPTION
**Pull request by NilsJacobsen**
Hey folks, 
new update on the json plugin.

- fix serializing bug #21 
- add formatting support for serialized output

New plugin settings (stays like the old one):
```js
export type PluginSettings = {
  pathPattern: string;
  variableReferencePattern?: [string, string];
};
```

### CODE REVIEW

File: src/plugin.ts [hunk 1]
```diff
@@ -250,6 +206,88 @@ function parseMessage(
...
+  ];
+
+  for (const { spacing, regex } of patterns) {
+    if (regex.test(jsonString)) {
+      return spacing;
+    }
+  }
+
+  return 2; // No matching spacing configuration found
+}
```
Comment (samuelstroschein) [inline src/plugin.ts:227]: @NilsJacobsen What a great idea!!!! 

If we can detect the spacing used when parsing, do we need the `serialize` setting at all?

Comment (NilsJacobsen) [reply src/plugin.ts:227]: nope it's already removed

Comment (samuelstroschein) [reply src/plugin.ts:227]: Ah so the type
 
```
export type PluginSettings = {
  pathPattern: string;
  variableReferencePattern?: [string, string];
  serialize?: {
    space?: Parameters<typeof JSON.stringify>[2];
    flatten?: boolean;
  }
};
```

is outdated?

Comment (samuelstroschein) [reply src/plugin.ts:227]: I was wondering if we couldn't get rid of `serialize` entirely. Seems like we can store every information in the metadata on parsing.

Comment (NilsJacobsen) [reply src/plugin.ts:227]:
```js
export type PluginSettings = {
  pathPattern: string;
  variableReferencePattern?: [string, string];
};
```
That's the new/old one

**Pull request merged**